packages/eas-cli/src/credentials/ios/actions/new/ConfigureProvisioningProfile.ts

```diff
@@ -0,0 +1,123 @@
+import assert from 'assert';
+import chalk from 'chalk';
+import ora from 'ora';
+
+import {
+  AppleDistributionCertificateFragment,
+  AppleProvisioningProfileFragment,
+} from '../../../../graphql/generated';
+import Log from '../../../../log';
+import { Action, CredentialsManager } from '../../../CredentialsManager';
+import { Context } from '../../../context';
+import { AppLookupParams } from '../../api/GraphqlClient';
+import { AppleProvisioningProfileMutationResult } from '../../api/graphql/mutations/AppleProvisioningProfileMutation';
+import { ProvisioningProfileStoreInfo } from '../../appstore/Credentials.types';
+import { AuthCtx } from '../../appstore/authenticate';
+import {
+  MissingCredentialsNonInteractiveError,
+  ProvisioningProfileNotFoundOnAppleServersError,
+} from '../../errors';
+
+export class ConfigureProvisioningProfile implements Action {
+  private _configuredProvisioningProfile?: AppleProvisioningProfileMutationResult;
+  constructor(
+    private app: AppLookupParams,
+    private distributionCertificate: AppleDistributionCertificateFragment,
+    private originalProvisioningProfile: AppleProvisioningProfileFragment
+  ) {}
+
+  public get provisioningProfile(): AppleProvisioningProfileMutationResult {
+    assert(
+      this._configuredProvisioningProfile,
+      'provisioningProfile can be accessed only after calling .runAsync()'
+    );
+    return this._configuredProvisioningProfile;
+  }
+
+  public async runAsync(_manager: CredentialsManager, ctx: Context): Promise<void> {
+    if (ctx.nonInteractive) {
+      throw new MissingCredentialsNonInteractiveError(
+        'Configuring Provisioning Profiles is only supported in interactive mode.'
+      );
+    }
+    const { developerPortalIdentifier, provisioningProfile } = this.originalProvisioningProfile;
+    if (!developerPortalIdentifier && !provisioningProfile) {
+      Log.warn("The provisioning profile we have on file cannot be validated on Apple's servers.");
+      return;
+    }
+
+    if (ctx.appStore.authCtx) {
+      const profilesFromApple = await ctx.appStore.listProvisioningProfilesAsync(
+        this.app.bundleIdentifier
+      );
+      const [matchingProfile] = profilesFromApple.filter(appleInfo =>
+        developerPortalIdentifier
+          ? appleInfo.provisioningProfileId === developerPortalIdentifier
+          : appleInfo.provisioningProfile === provisioningProfile
+      );
+      if (!matchingProfile) {
```

Comment:
Shouldn't we just create it at this point if we already logged in to apple?

Reply:
I've changed this so that `runAsync` returns `null` in a recoverable situation. SetupProvisioningProfile will call ConfigureProvisioningProfile, and if it returns `null`, we will try to create a new Provisioning Profile. Example of this are if the current Provisioning Profile doesnt exist on Apple Portal anymore. 

In a non recoverable situation, we throw an error and SetupProvisioningProfile will fail. Examples of this are if we try to run in non interactive mode, and we don't have all the build credentials setup.